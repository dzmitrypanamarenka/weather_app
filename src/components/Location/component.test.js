import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import { InfoWindow, Map, Marker } from 'google-maps-react';

import { Location, Message } from '../';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const store = mockStore({
  mapConfig: {
    coords: {
      lat: 52.520007,
      lng: 13.404954,
    },
    zoom: 10,
  },
  mapInfo: { visibility: true },
});
const LocationWithStore = (
  <Provider store={store}>
    <MemoryRouter>
      <Location/>
    </MemoryRouter>
  </Provider>
);

describe('Snapshot test', () => {
  it('should match snapshot', () => {
    const tree = renderer
      .create(LocationWithStore)
      .toJSON();
    expect(tree)
      .toMatchSnapshot();
  });
});

describe('Component render', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(LocationWithStore);
  });

  it('should render Location component', () => {
    expect(wrapper.find('.container').length)
      .toEqual(1);
  });
  it('should contains mounted components', () => {
    expect(wrapper.containsAllMatchingElements([
      Map,
      Marker,
      InfoWindow,
      Message,
    ]))
      .toEqual(true);
  });
});

