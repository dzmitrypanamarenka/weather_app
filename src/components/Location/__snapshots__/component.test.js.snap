// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test should match snapshot 1`] = `
<div
  className="container"
>
  <div
    className="jumbotron"
  >
    <h1
      className="title"
    >
      Hi, fellow!
    </h1>
    <p
      className="text"
    >
      Check out my amazing weather app. Buckle up and let your browser find your geoposition!
    </p>
    <button
      className="btn btn-primary"
      disabled={false}
      type="button"
    >
      <a
        className="link -weather"
        href="weather"
        onClick={[Function]}
      >
        Forecast
      </a>
    </button>
  </div>
  <div
    className="wrap"
    id="map"
  >
    <div>
      <div
        className={undefined}
        style={
          Object {
            "height": "100%",
            "position": "absolute",
            "width": "100%",
          }
        }
      >
        <div
          style={
            Object {
              "bottom": 0,
              "display": "inherit",
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          Loading map...
        </div>
      </div>
      <div />
    </div>
  </div>
</div>
`;
